### PR TITLE
Use dynamic learning rate decay for convergence

### PR DESCRIPTION
**Pull request by begeekmyfriend**
The evaluation sounds better than that with fixed learning rate.

Signed-off-by: begeekmyfriend <begeekmyfriend@gmail.com>

### CODE REVIEW

**Comment by seungwonpark** [comment]
Hi, your code looks great, and thanks for kindly sending PR!
Can you please show the audio samples you got (w/ the number of epochs) for comparison?

**Comment by begeekmyfriend** [comment]
[melgan_eval_mandarin.zip](https://github.com/seungwonpark/melgan/files/3982746/melgan_eval_mandarin.zip)
I have synthesized voices from 4 anchors (1 male and 3 females). And the checkpoint is only at epoch 375 and still under training. I think it helps convergence.

File: utils/train.py [hunk 1]
```diff
@@ -13,18 +12,35 @@
 from .validation import validate
 
 
+def cosine_decay(init_val, final_val, step, decay_steps):
+    alpha = final_val / init_val
```
Comment (delip) [inline utils/train.py:16]: Shouldn't this be `init_val / final_val`?

Comment (begeekmyfriend) [reply utils/train.py:16]: The learning rate decays. You might write a demo for testing.
```python
init_val = 1e-4
final_val = 1e-5
```

Comment (casper-hansen) [reply utils/train.py:16]: According to the following source it's **"Minimum learning rate value as a fraction of learning_rate."**
https://docs.w3cub.com/tensorflow~python/tf/train/cosine_decay/

Given the values, it looks like it's correct. The naming is just off - it should be the smallest value in the numerator and largest value in the denominator.

**Comment by bob80333** [comment]
Is this different from pytorch's built-in [CosineAnnealingLR?](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.CosineAnnealingLR)

**Comment by Liujingxiu23** [comment]
@begeekmyfriend  
Your tried different lr and found that the cos-lr is the best? And why it is suitable in melgan?
I am confused, when should we use unchanged lr, when should we use lr with decline，for example exponential decline in tacotron, and when shoud we use cos-lr.

**Comment by begeekmyfriend** [comment]
It is just a preference. Pick it or other as you like.

**Comment by Liujingxiu23** [comment]
@begeekmyfriend  Thank you for your quick reply. I used your branch of tacotron and found it is one of the best among a lot of code branchs. I will try the cos-lr as well as the apex in tfgan.
